index.html

```diff
@@ -4669,105 +4714,130 @@ <h3>Data First Approaches</h3>
+"statusReason": {
+  "value": "<span class="highlight">إلغاء</span>",
+  "lang: "<code>ar</code>"
+  "dir": "<code>rtl</code>"
+}
       </pre>
 
       <p>
-If the JSON-LD Context states that any value associated with the
-<code>nameHtml</code> property is of type <code>rdf:HTML</code>, then software
-agents have sufficient information to deterministically identify the text
-direction of the language.
+Utilization of the design pattern above assumes that the JSON-LD
+<a href="#extensibility">extension</a> associated with the
+<a>property</a> utilizes the Scoped Context feature of JSON-LD to
+alias the <code>@value</code>, <code>@language</code>, and
+<code>@direction</code> keywords to <code>value</code>, <code>lang</code>,
+and <code>dir</code>, respectively. An example of a JSON-LD Context snippet
+doing so is shown below:
       </p>
 
+      <pre class="example nohighlight" title="Specifying scoped aliasing for language information">
+"statusReason": {
+  <span class="highlight">"@context": {"value": "@value", "lang": "@language", "dir": "@direction"}</span>,
+  "@id": "https://www.w3.org/2018/credentials/examples#statusReason"
+}
+      </pre>
+
     </section>
 
     <section class="informative">
-      <h3>Ruby Text</h3>
+      <h3>Complex Language Markup</h3>
+
       <p>
-In some situations it is common to use
-<a href="https://en.wikipedia.org/wiki/Ruby_character">Ruby characters</a>
-to ensure text information is clear and as useful as possible.
-[[!JSON-LD]] support for HTML data allows text to be encoded with ruby,
-supporting these cases.
+When multiple languages, base directions, and/or annotations are used in a
+single natural language string, more complex mechanisms are typically required.
+It is possible to utilize markup languages, such as HTML, to encode
+text with multiple language and/or base directions. It is also
+possible to use the <code>rdf:HTML</code> datatype to encode such values
+accurately in JSON-LD.
+      </p>
+
+      <p>
+Implementers are strongly discouraged from encoding information as HTML
+because doing so 1) requires some version of an HTML processor, which
+increases the burden of processing language and/or base direction information,
+and 2) increases the security attack surface when utilizing this data model
+as blindly processing HTML could result in executing a <code>script</code>
+tag that an attacker injected at some point during the data production process.
       </p>
-    </section>
 
-    <section class="informative">
-      <h3>Further Information</h3>
       <p>
-The W3C Internationalisation document [[string-meta]] gives more information
-about the need to be able to provide reliable metadata about text to support
-internationalization.
+Implementers considering the use of HTML to encode complex language and/or
+base direction information might consider deconstructing the data into a
+format that does not require complex markup, such as an array of elements
```

Comment:
This advice worries me, since it could lead people to think that they should break a sentence up into separate strings everywhere the direction changes.  That would cause problems for translation of the string (see [Working with Composite Messages](https://www.w3.org/International/articles/composite-messages/)), not to mention the fact that an overall base direction (for the whole string) is still necessary, in order to pull together the individual parts in the correct order.

I'm not sure that much can be done for language, without markup.

But for base direction, the answer for _plain text_ strings is to use the Unicode formatting characters (see [How to use Unicode controls for bidi text](https://www.w3.org/International/questions/qa-bidi-unicode-controls)).

Reply:
+1 to this.

Most "multilingual" text actually has a base language. It's true that it is possible to write a sentence that demonstrates how language identification is needed within a single string (particularly when it comes to font selection or language-specific styling). But this is a corner case and is the sort of thing that calls for markup in any event.

Reply:
Done. Removed the bad advice in ea43b2513c9d9b31b6be740a7c52c336256f570b.